l1-contracts/test/foundry/unit/concrete/governance/Governance/PermanentRestriction.t.sol

```diff
@@ -0,0 +1,379 @@
+        chainAdmin.addRestriction(address(permRestriction));
+
+        Call memory call = Call({
+            target: hyperchain,
+            value: 0,
+            data: abi.encodeWithSelector(IAdmin.setPendingAdmin.selector, address(chainAdmin))
+        });
+
+        vm.startPrank(newChainAdmin);
+        permRestriction.validateCall(call, owner);
+        vm.stopPrank();
+    }
+
+    function test_validateCallNotValidatedSelector() public {
+        Call memory call = Call({
+            target: hyperchain,
+            value: 0,
+            data: abi.encodeWithSelector(IAdmin.acceptAdmin.selector)
+        });
+
+        vm.startPrank(newChainAdmin);
+        permRestriction.validateCall(call, owner);
+        vm.stopPrank();
+    }
+
+    function test_validateCallCallNotAllowed() public {
+        vm.prank(owner);
+        permRestriction.setSelectorIsValidated(IAdmin.acceptAdmin.selector, true);
+        Call memory call = Call({
+            target: hyperchain,
+            value: 0,
+            data: abi.encodeWithSelector(IAdmin.acceptAdmin.selector)
+        });
+
+        vm.expectRevert(abi.encodeWithSelector(CallNotAllowed.selector, call.data));
+
+        vm.startPrank(newChainAdmin);
+        permRestriction.validateCall(call, owner);
+        vm.stopPrank();
+    }
+
+    function test_validateCall() public {
+        vm.prank(owner);
+        permRestriction.setSelectorIsValidated(IAdmin.acceptAdmin.selector, true);
+        Call memory call = Call({
+            target: hyperchain,
+            value: 0,
+            data: abi.encodeWithSelector(IAdmin.acceptAdmin.selector)
+        });
+
+        vm.prank(owner);
+        permRestriction.setAllowedData(call.data, true);
+
+        vm.startPrank(newChainAdmin);
+        permRestriction.validateCall(call, owner);
+        vm.stopPrank();
+    }
+
+    function getDiamondCutData(address _diamondInit) internal returns (Diamond.DiamondCutData memory) {
+        InitializeDataNewChain memory initializeData = Utils.makeInitializeDataForNewChain(testnetVerifier);
+
+        bytes memory initCalldata = abi.encode(initializeData);
+
+        return Diamond.DiamondCutData({facetCuts: facetCuts, initAddress: _diamondInit, initCalldata: initCalldata});
+    }
+
+    function createNewChain(Diamond.DiamondCutData memory _diamondCut) internal {
+        vm.stopPrank();
+        vm.startPrank(0x0000000000000000000000000000000000000000);
+        iBridgehub.addStateTransitionManager(address(chainContractAddress));
+        iBridgehub.addToken(baseToken);
+        iBridgehub.setSharedBridge(sharedBridge);
+        iBridgehub.createNewChain({
+            _chainId: chainId,
+            _stateTransitionManager: address(chainContractAddress),
+            _baseToken: baseToken,
+            _salt: 0,
+            _admin: newChainAdmin,
+            _initData: abi.encode(_diamondCut)
+        });
+    }
+
+    function gettersSelectors() public pure returns (bytes4[] memory) {
```

Comment:
can't these be reused from other files? creating a chain seems like a common task

Reply:
I got rid off getGettersSelectors and added missing selector to utils function.
Regarding createNewChain i have not found a function that would use bridgehubs create new chain.
I can reuse getDiamondCutData function from StateTransitionManagerTest but would need to rename its setup function and make adjustments in tests inheriting from it.

Reply:
okay